crypto/evp/keymgmt_meth.c

```diff
@@ -280,3 +207,81 @@ void EVP_KEYMGMT_names_do_all(const EVP_KEYMGMT *keymgmt,
     if (keymgmt->prov != NULL)
         evp_names_do_all(keymgmt->prov, keymgmt->name_id, fn, data);
 }
+
+/*
+ * Internal API that interfaces with the method function pointers
+ */
+void *evp_keymgmt_newdata(const EVP_KEYMGMT *keymgmt)
+{
+    void *provctx = ossl_provider_ctx(EVP_KEYMGMT_provider(keymgmt));
+
+    /*
+     * TODO(3.0) 'new' is currently mandatory on its own, but when new
+     * constructors appear, it won't be quite as mandatory, so we have
+     * a check for future cases.
+     */
+    if (keymgmt->new == NULL)
+        return NULL;
+    return keymgmt->new(provctx);
+}
+
+void evp_keymgmt_freedata(const EVP_KEYMGMT *keymgmt, void *keydata)
+{
+    /* This is mandatory, no need to check for its presence */
+    keymgmt->free(keydata);
+}
+
+int evp_keymgmt_get_params(const EVP_KEYMGMT *keymgmt, void *keydata,
+                           OSSL_PARAM params[])
+{
+    if (keymgmt->get_params == NULL)
+        return 1;
+    return keymgmt->get_params(keydata, params);
+}
+
+const OSSL_PARAM *evp_keymgmt_gettable_params(const EVP_KEYMGMT *keymgmt)
+{
+    if (keymgmt->gettable_params == NULL)
+        return NULL;
+    return keymgmt->gettable_params();
+}
+
+int evp_keymgmt_has(const EVP_KEYMGMT *keymgmt, void *keydata, int selection)
+{
+    /* This is mandatory, no need to check for its presence */
+    return keymgmt->has(keydata, selection);
+}
+
+int evp_keymgmt_validate(const EVP_KEYMGMT *keymgmt, void *keydata,
+                         int selection)
+{
+    return keymgmt->validate(keydata, selection);
```

Comment:
this doesnt look mandatory so needs the NULL check?

Reply:
Ah, we get back to this that I was pondering a few days ago and then promptly forgot.

Thank you.

So, the question is, what should it return when the implementation hasn't supplied a validation function?  Between TRUE and FALSE, I haven't been able to decide...

Another option is to return another value that indicates that the validation function isn't present (unsupported).

What are your thoughts on that?

Reply:
I dont know if there is a good answer.. If it doesnt have a method then maybe return 1. Not sure we want to head down the return -2 path quite yet..

i.e: we assume its valid..

Reply:
Okie, I'll go with that

Reply:
Fixed (and the other ones that didn't check as well)